signer/src/signer/zone.c

```diff
@@ -1083,6 +1084,7 @@ zone_recover2(zone_type* zone)
                 "create task", zone_str, zone->name);
             goto recover_error2;
         }
+        task->lock = &zone->zone_lock;
```

Comment:
This makes me nervous. Following tasks MAY not get the same lock. At least in theory. It is also a recipe for deadlock when trying to grab zone->zone_lock inside an executed task.Is this safe?